pkg/controller/nodes/nodes_controller.go

```diff
@@ -288,3 +351,85 @@ func (r *ReconcileNodes) sendMarkedForTermination(oa *dynatracev1alpha1.OneAgent
 		},
 	})
 }
+
+func (r *ReconcileNodes) reconcileUnschedulableNode(node *corev1.Node, cache *Cache) error {
+	oneAgent, err := r.determineOneAgentForNode(node.Name)
+	if err != nil {
+		return err
+	}
+	if oneAgent == nil {
+		return nil
+	}
+
+	// determineOneAgentForNode  only returns a oneagent object if a node instance is present
+	instance, _ := oneAgent.Status.Instances[node.Name]
+	cachedNode, err := cache.Get(node.Name)
+	if err != nil {
+		if err == ErrNotFound {
```

Comment:
I'm curious here. Is there any scenario where an Update notification would be processed after a Deletion? It would re-add the entry and cause for the message to be sent twice.

I guess if a Update notification came while reconcileAll was running, but before listing all nodes.

Shouldn't break anything however. I know it was mentioned that we currently see the messages sent twice occasionally, but the timing needed here shouldn't happen that often even.

On the other hand, if I'm not mistaken, the case handled here would be when we get an update for an entry before it was caught up on reconcileAll() but for that to happen the Node needs to have been lived for less than 5 minutes before being deleted.

Don't think we need to concern ourselves with this scenario. Or perhaps is there another case handled here?